tests/test_treedb_cli.py

```
@@ -324,13 +329,30 @@ def search_vector_index(self, *args, **kwargs):
+    db = object.__new__(TreeDB)
+    db._client = None
+    db._new_client = lambda: client
+
+    with db.init():
+        assert db.client is client
+
+    assert client.closed is True
+    assert db._client is None
+
+
 def _tree_db_for_response(search_param: dict, response):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_tree_db_for_response`

(ANN202)